Doc/c-api/module.rst

```diff
@@ -609,6 +634,19 @@ state:
 
    .. versionadded:: 3.9
 
+.. c:function:: int PyModule_ExperimentalSetGIL(PyObject *module, void *gil)
```

Comment:
Sorry, I'll get a PR ready. I went with this name because having both `Unstable` and `Experimental` in the name felt redundant, and in [this comment](https://github.com/python/cpython/pull/116882#discussion_r1586715045), @ericsnowcurrently expressed a preference for `PyModule_ExperimentalSetGIL()` over `PyUnstable_Module_SetGIL()`.

Does anyone else have strong feelings on whether it should be `PyUnstable_Module_ExperimentalSetGIL()` or `PyUnstable_Module_SetGIL()`?

Reply:
PR here: https://github.com/python/cpython/pull/118645

IMO, the “experimental” is redundant. If we need “Experimental” too, we should rethink the entire PEP-689 naming convention.